pkg/entities/quest.go

```diff
@@ -294,6 +352,22 @@ func (e *Entity) ClaimQuestsRewards(req request.ClaimQuestsRewardsRequest) (*res
 		IsClaimed:   &claimed,
 		QuestID:     req.QuestID,
 	}
+
+	//validate quest id
+	err_id := e.validateQuestID(listQ.QuestID)
+	// validate user
+	err_user := e.validateUser(*listQ.UserID)
+
+	if err_id != nil && err_user == nil {
+		return nil, err_id
+	}
+	if err_id == nil && err_user != nil {
+		return nil, err_user
+	}
+	if err_id != nil && err_user != nil {
+		return nil, errors.New(err_id.Error() + ", " + err_user.Error())
+	}
+
```

Comment:
same as above

Reply:
Same as above, this endpoint will return data: null if user failed to call API.